org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/StubUtility.java

```diff
@@ -713,7 +713,12 @@ public static String getMethodComment(ICompilationUnit cu, String typeName, Meth
 		context.setVariable(CodeTemplateContextType.ENCLOSING_TYPE, typeName);
 		context.setVariable(CodeTemplateContextType.ENCLOSING_METHOD, decl.getName().getIdentifier());
 		if (!decl.isConstructor()) {
-			context.setVariable(CodeTemplateContextType.RETURN_TYPE, ASTNodes.asString(getReturnType(decl)));
+			ASTNode retType= getReturnType(decl);
```

Comment:
Unfortunately that won't work. Line 730+ reads as follows:
```
		TemplateBuffer buffer;
		try {
			buffer= context.evaluate(template);
		} catch (BadLocationException | TemplateException e) {
			throw new CoreException(Status.CANCEL_STATUS);
		}
```

This will throw an AssertionFailedException.

Reply:
To be more clear, the first thing CodeTemplateContext.evaluate(template) does is verify that all required attributes are set:

```
	@Override
	public TemplateBuffer evaluate(Template template) throws BadLocationException, TemplateException {
		// test that all variables are defined
		Iterator<TemplateVariableResolver> iterator= getContextType().resolvers();
		while (iterator.hasNext()) {
			TemplateVariableResolver var= iterator.next();
			if (var instanceof CodeTemplateContextType.CodeTemplateVariableResolver) {
				Assert.isNotNull(getVariable(var.getType()), "Variable " + var.getType() + "not defined"); //$NON-NLS-1$ //$NON-NLS-2$
			}
		}
```

Reply:
It doesn't flag constructors:

You could set the template context to constructor above:

		if (decl.isConstructor()) {
			templateName= CodeTemplateContextType.CONSTRUCTORCOMMENT_ID;

by changing the if to:  if (getReturnType() == null) {

similarly with the check below: if (!getReturnType() == null)